### PR TITLE
Increase coverage of lightweight unit tests

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
Adds unit tests for `MQTT_SerializePublishHeader`, `MQTT_SerializeAck`, `MQTT_GetDisconnectPacketSize`, and `MQTT_GetPingreqPacketSize`. Branch coverage of mqtt_lightweight.c is 100%.

This should not be merged until #957 is.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 1]
```diff
@@ -712,42 +712,24 @@ static MQTTStatus_t processPublishFlags( uint8_t publishFlags,
...
 {
+    const char * const pConnackResponses[ 6 ] =
+    {
+        "Connection accepted.",                               /* 0 */
+        "Connection refused: unacceptable protocol version.", /* 1 */
+        "Connection refused: identifier rejected.",           /* 2 */
+        "Connection refused: server unavailable",             /* 3 */
+        "Connection refused: bad user name or password.",     /* 4 */
+        "Connection refused: not authorized."                 /* 5 */
+    };
```
Comment (muneebahmed10) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:723]: Originally this function was like this, but in a macro guard so that is would only be allocated if logging was enabled. #937 converted this into a switch statement, since that macro is no longer defined anywhere. This reverts it back, since part of that switch was unreachable, and I'm going to trust the compiler to not allocate memory for unused variables.

File: libraries/standard/mqtt/include/mqtt_lightweight.h [hunk 2]
```diff
@@ -478,10 +478,6 @@ MQTTStatus_t MQTT_GetPingreqPacketSize( size_t * pPacketSize );
  */
 MQTTStatus_t MQTT_SerializePingreq( const MQTTFixedBuffer_t * const pBuffer );
 
-MQTTStatus_t MQTT_GetIncomingPacket( MQTTTransportRecvFunc_t recvFunc,
```
Comment (leegeth) [inline libraries/standard/mqtt/include/mqtt_lightweight.h:481]: Why is this removed. I think we are planning to implement this function.

File: libraries/standard/mqtt/src/mqtt_lightweight.c [hunk 3]
```diff
@@ -1929,15 +1926,6 @@ MQTTStatus_t MQTT_SerializePingreq( const MQTTFixedBuffer_t * const pBuffer )
...
 
-MQTTStatus_t MQTT_GetIncomingPacket( MQTTTransportRecvFunc_t recvFunc,
-                                     MQTTNetworkContext_t networkContext,
-                                     MQTTPacketInfo_t * const pIncomingPacket )
-{
-    return MQTTSuccess;
-}
-
-/*-----------------------------------------------------------*/
-
```
Comment (leegeth) [inline libraries/standard/mqtt/src/mqtt_lightweight.c:1927]: same comment as above.

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c [hunk 4]
```diff
@@ -1570,6 +1606,67 @@ void test_MQTT_SerializePublishHeader( void )
...
+    expectedPacket[ 0 ] = packetType;
+    expectedPacket[ 1 ] = 2U;
+    expectedPacket[ 2 ] = PACKET_ID >> 8;
+    expectedPacket[ 3 ] = PACKET_ID & UINT8_MAX;
+
+    /* Verify parameters. */
+    status = MQTT_SerializeAck( NULL, packetType, PACKET_ID );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    status = MQTT_SerializeAck( &fixedBuffer, packetType, 0 );
```
Comment (leegeth) [inline libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1684]: missed validation?

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 2 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:481]: No, it's not in our API doc, and we wrote MQTT_GetIncomingPacketTypeAndLength instead. I believe we intended to remove it before, but never got around to it.

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 4 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1684]: Are you referring to the reason for the expected failure? I can add in a comment that packet ID 0 is not allowed.

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c [hunk 5]
```diff
@@ -968,6 +1009,15 @@ void test_MQTT_SerializePublish( void )
...
     TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    memset( ( void * ) expectedPacket, 0x00, sizeof( expectedPacket ) );
+    pIterator = expectedPacket;
+    /* Dup = 0x8, QoS2 = 0x4, Retain = 0x1. 8 + 4 + 1 = 0xD. */
+    *pIterator++ = MQTT_PACKET_TYPE_PUBLISH | 0xD;
+    pIterator += encodeRemainingLength( pIterator, remainingLength );
+    pIterator += encodeString( pIterator, publishInfo.pTopicName, publishInfo.topicNameLength );
+    *pIterator++ = UINT16_HIGH_BYTE( PACKET_ID );
+    *pIterator++ = UINT16_LOW_BYTE( PACKET_ID );
+    TEST_ASSERT_EQUAL_MEMORY( expectedPacket, buffer, packetSize );
```
Comment (leegeth) [inline libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1020]: I think it would add value if we can add the buffer validation as part of all the Success cases. Line 976. What are your thoughts?

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 2 above)
Comment (leegeth) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:481]: I think the plan is to still keep it and implement it. It is an additional API to provide the customer the choice of avoiding multiple transport reads. It came up in the design proposal.

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 4 above)
Comment (leegeth) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1684]: Did you miss yo do this step here? TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 5 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1020]: I'm not sure I understand. Nothing is written into the buffer by line 976, that's just making sure the calculated packet size does not exceed the buffer size

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 4 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1684]: Oh, I didn't see that. Yes

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 5 above)
Comment (leegeth) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1020]: Sorry, wrong line number, I meant 982

Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1020]: That buffer is verified in line 993. Everything between 982 and 993 just generates the expected packet. Should it be rearranged so the buffer verification is closer to when the status is tested?

Comment (leegeth) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1020]: Yea. Ok. I see the problem. I missed to notice the assert in 993 and was thinking that it is setting the expected buffer for the next test. I think it is ok as it is.

File: libraries/standard/mqtt/include/mqtt_lightweight.h (hunk 2 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/include/mqtt_lightweight.h:481]: After some offline discussion it does seem like that is the plan, so I will re add this function.

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c [hunk 6]
```diff
@@ -1542,6 +1632,178 @@ void test_MQTT_GetIncomingPacketTypeAndLength( void )
...
+
+    /* Success case. */
+    status = MQTT_GetPublishPacketSize( &publishInfo, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    status = MQTT_SerializePublishHeader( &publishInfo,
+                                          PACKET_ID,
+                                          remainingLength,
+                                          &fixedBuffer,
+                                          &headerSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
```
Comment (leegeth) [inline libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1727]: missed to verify the serialized buffer?

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c [hunk 7]
```diff
@@ -1542,6 +1632,178 @@ void test_MQTT_GetIncomingPacketTypeAndLength( void )
...
+    publishInfo.pPayload = "test";
+    publishInfo.payloadLength = 4;
+    status = MQTT_GetPublishPacketSize( &publishInfo, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    status = MQTT_SerializePublishHeader( &publishInfo,
+                                          0,
+                                          remainingLength,
+                                          &fixedBuffer,
+                                          &headerSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
```
Comment (leegeth) [inline libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1739]: missed to verify the serialized buffer?

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 6 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1733]: Yes, added

File: libraries/standard/mqtt/utest/mqtt_lightweight_utest.c (hunk 7 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/utest/mqtt_lightweight_utest.c:1762]: Added
